### PR TITLE
Reduce calls to getQuestionDetails and getStudentProfilePictureLink #4114

### LINKED ISSUE #4114: Reduce calls to getQuestionDetails and getStudentProfilePictureLink
The repeated getQuestionDetails were causing the bulk of the slowdown on the .java side of things. The repeated getStudentProfilePictureLink were slow even on the old pure jsp files. 

On a page that took 6s to load on the old page, the above improvements reduced the page load times from ~25s to about 8s. Now the bulk of the slowdown is caused by the tag files.

### PR DESCRIPTION
**Pull request by kanghj**
Fixes #4114

### CODE REVIEW

**Comment by damithc** [comment]
@yj-soh can you review soon?

File: src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java [hunk 1]
```diff
@@ -74,6 +76,9 @@
     // giver > recipient > question, recipient > giver > question
     private LinkedHashMap<String, InstructorFeedbackResultsSectionPanel> sectionPanels;
     
+    private Map<FeedbackQuestionAttributes, FeedbackQuestionDetails> questionToDetailsMap = new HashMap<>();
+    private Map<String, String> profilePictureLink = new HashMap<>();
```
Comment (yj-soh) [inline src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java]: Maybe make this plural or suffix with `Map`?

File: src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java [hunk 2]
```diff
@@ -1278,12 +1291,12 @@ private void configureResponseRow(FeedbackQuestionAttributes question,
         }
     }
     
-    private Url getProfilePictureIfEmailValid(String email) {
+    private String getProfilePictureIfEmailValid(String email) {
```
Comment (yj-soh) [inline src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java]: If `Url` should be used but not now, maybe add a comment?

File: src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java [hunk 3]
```diff
@@ -1636,10 +1649,15 @@ private boolean isResponseVisibleTo(FeedbackParticipantType participantType, Fee
...
     }
     
-    // TODO remove this entirely and use PageData method directly
     public String getProfilePictureLink(String studentEmail) {
-        return getStudentProfilePictureLink(StringHelper.encrypt(studentEmail),
-                                            StringHelper.encrypt(instructor.courseId));
+        if (!profilePictureLink.containsKey(studentEmail)) {
+            profilePictureLink.put(studentEmail, 
+                                   getStudentProfilePictureLink(StringHelper.encrypt(studentEmail),
+                                                                StringHelper.encrypt(instructor.courseId)));
```
Comment (yj-soh) [inline src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java]: Should this override `PageData.getStudentProfilePictureLink()`?
Also, considering it's only used here, the method should be private.
As for the removed comment, should it be replaced with another comment targeting L1655 / 1656 directly or is the current behaviour correct and expected now?

File: src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java [hunk 4]
```diff
@@ -1686,11 +1704,19 @@ public int getStartIndex() {
...
+        }
+        
+        return displayableCourseId;
     }
 
     public String getFeedbackSessionName() {
-        return sanitizeForHtml(bundle.feedbackSession.feedbackSessionName);
+        if (displayableFsName == null) {
+            displayableFsName = sanitizeForHtml(bundle.feedbackSession.feedbackSessionName);
+        }
```
Comment (yj-soh) [inline src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java]: Should these be done in `initCommonVariables()`?
I'm not sure how many times these are called on the front-end, but it doesn't seem effective having to do an if check each time.

**Comment by kanghj** [comment]
Changes made @yj-soh

**Pull request merged**